bin/echo.c

```diff
@@ -175,9 +180,19 @@ int print_connection_info(struct s2n_connection *conn)
         printf("Application protocol: %s\n", s2n_get_application_protocol(conn));
     }
 
-    printf("Curve: %s\n", s2n_connection_get_curve(conn));
-    printf("KEM: %s\n", s2n_connection_get_kem_name(conn));
-    printf("KEM Group: %s\n", s2n_connection_get_kem_group_name(conn));
+    const char *kem = s2n_connection_get_kem_name(conn);
+    if (strcmp(kem, "NONE") != 0) {
+        printf("Legacy TLS1.2 KEM: %s (%s, but TLS1.3 PQ hybrid key exchange strongly recommended instead. "
+               "See https://github.com/aws/s2n-tls/blob/main/docs/usage-guide/topics/ch15-post-quantum.md)\n",
```

Comment:
I can't quite tell whether "legacy" refers to TLS 1.2 or the KEM. I think it's referring to the KEM in which case I was expecting the linked usage guide to explain why TLS 1.2 PQ was weaker/flawed relative to TLS 1.3, but it doesn't seem to include that information.

Could we either expand on that information here, or expand on it in the linked usage guide?

Reply:
The linked usage guide intentionally doesn't mention TLS1.2 support or list any security policies that would enable it. [This comment](https://github.com/aws/s2n-tls/pull/4677#discussion_r1698033816) discussed that choice.

I tried to very briefly explain the recommendation in the output message. While I dislike how long the message is, I prefer only telling people who are already using TLS1.2 PQ over advertising its existence to everyone with the usage guide :)